protocol/x/revshare/keeper/revshare.go

```diff
@@ -124,15 +125,17 @@ func (k Keeper) SetUnconditionalRevShareConfigParams(ctx sdk.Context, config typ
 }
 
 // ValidateRevShareSafety roughly checks if the total rev share is valid using the formula below
-// highest_affiliate_taker_share + sum(unconditional_rev_shares) + market_mapper_rev_share < 100%
-// Note: this is just an estimate as affiliate rev share is based on taker fees, while
-// the rest of the rev share is based on net fees.
-// TODO(OTE-788): Revisit this formula to ensure accuracy.
+// highest_affiliate_taker_share * (highest_taker_fee_share_ppm /
+// (lowest_maker_fee_share_ppm + highest_taker_fee_share_ppm))
+// + sum(unconditional_rev_shares) + market_mapper_rev_share < 100%
 func (k Keeper) ValidateRevShareSafety(
+	ctx sdk.Context,
 	affiliateTiers affiliatetypes.AffiliateTiers,
 	unconditionalRevShareConfig types.UnconditionalRevShareConfig,
 	marketMapperRevShareParams types.MarketMapperRevenueShareParams,
 ) bool {
+	highestTakerFeeSharePpm := k.feetiersKeeper.GetHighestTakerFee(ctx)
```

Comment:
Ideally we stick to the same pattern and pass in everything as input so this function is static. Since technically you want to call `ValidateRevShareSafety` before updating fee tiers too. Wdyt?

Reply:
Wrong variable name: this is just taker fee - not relevant to any fee share

Reply:
Agreed that it would be better to check fee tiers as well - dont want to rely on someone remembering these